presto-main/src/main/java/com/facebook/presto/type/FloatOperators.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.operator.scalar.ScalarOperator;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.type.StandardTypes;
+
+import static com.facebook.presto.metadata.OperatorType.NEGATION;
+import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
+
+public final class FloatOperators
+{
+    private FloatOperators()
+    {
+    }
+
+    @ScalarOperator(NEGATION)
+    @SqlType(StandardTypes.FLOAT)
+    public static long negate(@SqlType(StandardTypes.FLOAT) long value)
+    {
+        try {
+            Math.toIntExact(value);
+        }
+        catch (ArithmeticException e) {
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, "More than 32b value provided for Float type");
```

Comment:
The same as for `writeLong` method - what would you like to print here: decimal, double value? I think the same approach as for `writeLong` would be the best since you should never have any >32b number passed to this method.